src/Moq/MatcherFactory.cs

```diff
@@ -71,9 +71,23 @@ internal static class MatcherFactory
 					return new Pair<IMatcher, Expression>(new RefMatcher(constant.Value), constant);
 				}
 			}
-			else if (parameter.IsDefined(typeof(ParamArrayAttribute), true) && (argument.NodeType == ExpressionType.NewArrayInit || !argument.Type.IsArray))
```

Comment:
This last condition just plain didn't make any sense. Why would we want to enter into the `ParamArrayMatcher` clause for `params` parameters whose argument is *not* an array?! This may have been an attempt to allow `null` values in place of an empty array, but at least the C# compiler will not produce such a value.